[BUG]: inferred web middleware type depends on `import("undici-types")`

### What happened?

@octokit/webhooks v13.7.x (ie. versions featuring #1078) features the following emitted type pattern in `dist-types/middleware/web/index.d.ts`:

```ts
export declare function createWebMiddleware(webhooks: Webhooks, { path, log, }?: MiddlewareOptions): (request: Request) => Promise<import("undici-types").Response>;
```

The middleware return type is not specified in the source, and is inferred as `Promise<import("undici-types").Response>` rather than `Promise<Response>` for the declaration emit, via @types/node's fetch types.

Since undici-types is not a dependency of the octokit packages, this leads to the following TypeScript compilation error, depending on build environment:

```
error TS2307: Cannot find module 'undici-types' or its corresponding type declarations.
```

It looks like this has bricked a number of GitHub Actions workflows.

### Versions

@octokit/webhooks v13.7.x

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/src/middleware/web/middleware.ts
+++ b/src/middleware/web/middleware.ts
@@ -11,7 +11,7 @@ export async function middleware(
   webhooks: Webhooks,
   options: Required<MiddlewareOptions>,
   request: Request,
-) {
+): Promise<Response> {
   let pathname: string;
   try {
     pathname = new URL(request.url as string, "http://localhost").pathname;

